### PR TITLE
[REF] refactor test with error raising

### LINKED ISSUE #3796: refactor tests to use context manager
Tests that check for errors or warnings and should use context manager.

_Originally posted by @ymzayek in https://github.com/nilearn/nilearn/pull/3795#discussion_r1247566146_

Example:

This:

https://github.com/nilearn/nilearn/blob/8a7a80cbbb035e40270d35ff17b3022214bcf0db/nilearn/maskers/tests/test_multi_nifti_masker.py#L26-L32

Should become: 

```python
    # Check that if we have not fit the masker we get a intelligible
    # error
    with pytest.raises(ValueError):
        masker.transform([[img]])
```

video intros to:
- pytest: https://www.youtube.com/watch?v=mzlH8lp4ISA
- test errors with pytest: https://www.youtube.com/watch?v=6nRxZyQwwlE

See pytest documentation for more information: 
- https://docs.pytest.org/en/7.1.x/how-to/assert.html#assertions-about-expected-exceptions

list of files to to modify:

```bash
$ find nilearn -name '*.py' -exec grep -H "pytest.raises" {} \; | grep -v with

nilearn/datasets/tests/test_atlas.py:    pytest.raises(
nilearn/datasets/tests/test_atlas.py:    pytest.raises(ValueError, atlas.fetch_atlas_talairach, "bad_level")
nilearn/datasets/tests/test_atlas.py:    pytest.raises(ValueError, atlas.fetch_atlas_schaefer_2018, n_rois=44)
nilearn/datasets/tests/test_atlas.py:    pytest.raises(ValueError, atlas.fetch_atlas_schaefer_2018, yeo_networks=10)
nilearn/datasets/tests/test_atlas.py:    pytest.raises(ValueError, atlas.fetch_atlas_schaefer_2018, resolution_mm=3)
nilearn/datasets/tests/test_neurovault.py:    pytest.raises(requests.RequestException, neurovault._get_batch, "http://")
nilearn/datasets/tests/test_neurovault.py:        pytest.raises(
nilearn/datasets/tests/test_neurovault.py:    pytest.raises(ValueError, neurovault._get_batch, no_results_url)
nilearn/datasets/tests/test_neurovault.py:        pytest.raises(
nilearn/datasets/tests/test_neurovault.py:        pytest.raises(
nilearn/datasets/tests/test_neurovault.py:    pytest.raises(ValueError, neurovault.fetch_neurovault_ids, mode="bad")
nilearn/tests/test_signal.py:    pytest.raises(TypeError, nisignal.clean, x, low_pass=False)
nilearn/tests/test_signal.py:    pytest.raises(TypeError, nisignal.clean, x, high_pass=False)
nilearn/tests/test_signal.py:    pytest.raises(ValueError, clean, sx, low_pass=0.4, high_pass=0.5, t_r=2.5)
nilearn/tests/test_signal.py:    pytest.raises(TypeError, nisignal.clean, signals, confounds=1)
nilearn/tests/test_signal.py:    pytest.raises(ValueError, nisignal.clean, signals, confounds=np.zeros(2))
nilearn/tests/test_signal.py:    pytest.raises(ValueError, nisignal.clean, signals,
nilearn/tests/test_signal.py:    pytest.raises(ValueError, nisignal.clean, signals,
nilearn/tests/test_signal.py:    pytest.raises(ValueError, nisignal.clean, signals[:-1, :],
nilearn/tests/test_signal.py:    pytest.raises(TypeError, nisignal.clean, signals,
nilearn/tests/test_signal.py:    error_msg = pytest.raises(ValueError, nisignal.clean, signals,
nilearn/tests/test_signal.py:    pytest.raises(ValueError, nisignal.clean, signals, t_r=None,
nilearn/tests/test_signal.py:    pytest.raises(ValueError, nisignal.clean, signals,
nilearn/tests/test_signal.py:    pytest.raises(ValueError, nisignal.clean, signals,
nilearn/tests/test_signal.py:    pytest.raises(AssertionError, np.testing.assert_almost_equal,
nilearn/tests/test_masking.py:    pytest.raises(ValueError, compute_epi_mask, mean_image)
nilearn/tests/test_masking.py:    pytest.raises(ValueError, compute_background_mask, mean_image)
nilearn/tests/test_masking.py:    pytest.raises(ValueError, masking.apply_mask,
nilearn/tests/test_masking.py:    pytest.raises(ValueError, masking.apply_mask,
nilearn/tests/test_masking.py:    pytest.raises(ValueError, masking.apply_mask,
nilearn/tests/test_masking.py:    pytest.raises(ValueError, masking.apply_mask,
nilearn/tests/test_masking.py:    pytest.raises(TypeError, unmask, vec_1D, mask_img)
nilearn/tests/test_masking.py:    pytest.raises(TypeError, unmask, [vec_1D], mask_img)
nilearn/tests/test_masking.py:    pytest.raises(TypeError, unmask, vec_2D, mask_img)
nilearn/tests/test_masking.py:    pytest.raises(TypeError, unmask, [vec_2D], mask_img)
nilearn/tests/test_masking.py:    pytest.raises(TypeError, compute_multi_epi_mask, [])
nilearn/tests/test_masking.py:        pytest.raises(ValueError, compute_multi_epi_mask,
nilearn/tests/test_masking.py:    pytest.raises(TypeError, compute_multi_brain_mask, [])
nilearn/tests/test_masking.py:    pytest.raises(ValueError, compute_multi_brain_mask, imgs)
nilearn/tests/test_masking.py:    pytest.raises(TypeError, unmask, X, mask_img)
nilearn/tests/test_masking.py:    pytest.raises(TypeError, unmask, X, mask_img)
nilearn/surface/tests/test_surface.py:    pytest.raises(ValueError, surface._sample_locations,
nilearn/surface/tests/test_surface.py:    pytest.raises(ValueError, surface._projection_matrix,
nilearn/maskers/tests/test_multi_nifti_maps_masker.py:            pytest.raises(ValueError, masker11.transform, fmri12_img)
nilearn/maskers/tests/test_multi_nifti_maps_masker.py:            pytest.raises(ValueError, masker11.transform, fmri21_img)
nilearn/maskers/tests/test_multi_nifti_maps_masker.py:            pytest.raises(ValueError, masker11.fit)
nilearn/maskers/tests/test_multi_nifti_maps_masker.py:    pytest.raises(ValueError, masker11.fit)
nilearn/maskers/tests/test_multi_nifti_maps_masker.py:    pytest.raises(
nilearn/maskers/tests/test_multi_nifti_maps_masker.py:    pytest.raises(
nilearn/maskers/tests/test_multi_nifti_masker.py:    pytest.raises(
nilearn/maskers/tests/test_multi_nifti_masker.py:    pytest.raises(ValueError, masker.fit, img)
nilearn/maskers/tests/test_nifti_masker.py:        pytest.raises(TypeError, masker.fit)
nilearn/maskers/tests/test_nifti_masker.py:    pytest.raises(ValueError, masker.fit_transform, data_img)
nilearn/maskers/tests/test_multi_nifti_labels_masker.py:    pytest.raises(ValueError, masker11.transform, fmri12_img)
nilearn/maskers/tests/test_multi_nifti_labels_masker.py:    pytest.raises(ValueError, masker11.transform, fmri21_img)
nilearn/maskers/tests/test_multi_nifti_labels_masker.py:    pytest.raises(ValueError, masker11.fit)
nilearn/maskers/tests/test_multi_nifti_labels_masker.py:    pytest.raises(ValueError, masker11.fit)
nilearn/maskers/tests/test_multi_nifti_labels_masker.py:    pytest.raises(
nilearn/maskers/tests/test_multi_nifti_labels_masker.py:    pytest.raises(
nilearn/maskers/tests/test_nifti_maps_masker.py:            pytest.raises(ValueError, masker11.transform, fmri12_img)
nilearn/maskers/tests/test_nifti_maps_masker.py:            pytest.raises(ValueError, masker11.transform, fmri21_img)
nilearn/maskers/tests/test_nifti_maps_masker.py:            pytest.raises(ValueError, masker11.fit)
nilearn/maskers/tests/test_nifti_maps_masker.py:    pytest.raises(ValueError, masker11.fit)
nilearn/maskers/tests/test_nifti_maps_masker.py:    pytest.raises(ValueError, NiftiMapsMasker, maps33_img,
nilearn/maskers/tests/test_nifti_maps_masker.py:    pytest.raises(ValueError, NiftiMapsMasker, maps33_img,
nilearn/maskers/tests/test_nifti_labels_masker.py:    pytest.raises(ValueError, masker11.transform, fmri12_img)
nilearn/maskers/tests/test_nifti_labels_masker.py:    pytest.raises(ValueError, masker11.transform, fmri21_img)
nilearn/maskers/tests/test_nifti_labels_masker.py:    pytest.raises(ValueError, masker11.fit)
nilearn/maskers/tests/test_nifti_labels_masker.py:    pytest.raises(ValueError, masker11.fit)
nilearn/maskers/tests/test_nifti_labels_masker.py:    pytest.raises(ValueError, NiftiLabelsMasker, labels33_img,
nilearn/maskers/tests/test_nifti_labels_masker.py:    pytest.raises(ValueError, NiftiLabelsMasker, labels33_img,
nilearn/_utils/tests/test_param_validation.py:                pytest.raises(
nilearn/_utils/tests/test_numpy_conversions.py:    pytest.raises(ValueError, as_ndarray, "test string")
nilearn/_utils/tests/test_numpy_conversions.py:    pytest.raises(ValueError, as_ndarray, [], order="invalid")
nilearn/_utils/tests/test_numpy_conversions.py:        pytest.raises(TypeError, csv_to_array, filename, delimiters="?!")
nilearn/_utils/tests/test_ndimage.py:    pytest.raises(ValueError, largest_connected_component, a)
nilearn/_utils/tests/test_ndimage.py:    pytest.raises(ValueError, largest_connected_component, img)
nilearn/_utils/tests/test_ndimage.py:    pytest.raises(ValueError, largest_connected_component, "Test String")
nilearn/plotting/tests/test_img_plotting/test_plot_stat_map.py:    pytest.raises(ValueError, plot_stat_map, testdata_3d_for_plotting['img'],
nilearn/plotting/tests/test_img_plotting/test_plot_anat.py:    pytest.raises(Exception, plot_anat, display_mode="zzz")
nilearn/regions/tests/test_rena_clustering.py:    pytest.raises(ValueError, rena.fit, X)
nilearn/regions/tests/test_rena_clustering.py:        pytest.raises(ValueError, rena.fit, X)
```

```bash
$ find nilearn -name '*.py' -exec grep -H "pytest.warns" {} \; | grep -v with

nilearn/tests/test_signal.py:    pytest.warns(UserWarning, nisignal.clean, signals,
nilearn/plotting/tests/test_html_document.py:    pytest.warns(UserWarning, _open_views)
nilearn/plotting/tests/test_html_document.py:    pytest.warns(UserWarning, _open_views)
```

### PR DESCRIPTION
**Pull request by FrancoisPgm**
Closes #3796 .

Changes proposed in this pull request:

- the tests in the form `pytest.raises(<Error>, <func>, <args>)` were changed to use the `with pytest.raises(<Error>)` context manager

### CODE REVIEW

File: nilearn/surface/tests/test_surface.py
Comment (Remi-Gau) [inline nilearn/surface/tests/test_surface.py:1]: Seems that black was run on this file and 2 others when it should not have.

It is something a bit tricky about our current black set up at the moment that can leas this kind of outcome.

We can try to look how to fix that tomorrow.

**Review by Remi-Gau — CHANGES_REQUESTED** [review]
Changes look good (except for black going to town when it should not have) but you should:

-     add an entry to the changelog
-     add yourself to the list of contributors

see here: https://nilearn.github.io/dev/development.html#changelog

and ask us if you need guidance

**Comment by Remi-Gau** [comment]
> The failed flake8 test comes from a not yet blacked file, so not my fault :P

Can you try to fix the issues manually to make flake8 happy.

**Review by ymzayek — COMMENTED** [review]
Looks good! Just minor flake8 issues: https://github.com/nilearn/nilearn/actions/runs/5613902463/job/15210922744?pr=3854

**Comment by Remi-Gau** [comment]
That list of commit messages is going in the wrap up session!!!

I feel you so much on this.

**Review by ymzayek — APPROVED** [review]
Thx @FrancoisPgm! Flake8 is happy and so are we!

**Review by Remi-Gau — APPROVED** [review]
LGTM. Thanks for doing this @FrancoisPgm

**Comment by Remi-Gau** [comment]
Let me know if you want us to handle the merge conflict: it should be easy to fix but, ping us if you need help

**Comment by Remi-Gau** [comment]
Will fix it

**Pull request merged**